src/oop/programs_with_multiple_classes/hard/q3/MainTest.java

```diff
@@ -0,0 +1,364 @@
+        _assertEquals(count, book.getFieldValue(bookInstance, this.count),
+                "Your " + book + " constructor does not correctly initialize the " + this.count + " attribute.");
+        _assertEquals(type, book.getFieldValue(bookInstance, this.type),
+                "Your " + book + " constructor does not correctly initialize the " + this.type + " attribute.");
+    }
+
+    @Tag("dependency1")
+    @ParameterizedTest
+    @MethodSource("bookInputProvider")
+    public void bookHasCorrectIncrementBookCountMethod(int count, String type) throws Throwable {
+        Object[][] arguments = new Object[][]{
+                {count, int.class},
+                {type, String.class}
+        };
+        Object bookInstance = book.createInstance(arguments);
+        String incorrectIncrementBookCount = "Your %s method does not correctly increment and return the value of the %s attribute.";
+
+        assertTrue(book.hasMethod(bookMethodName, new Class[]{}),
+                String.format("Your %s class is missing the method %s.", bookClass, bookMethodName));
+        assertTrue(book.hasMethod(bookMethodName, new Class[]{}, int.class),
+                String.format("Your %s class %s method does not have the correct return type.", bookClass, bookMethodName));
+        assertTrue(book.hasMethod(bookMethodName, new Class[]{}, int.class, new String[]{"public"}),
+                String.format("Your %s class %s method does not have the correct visibility modifier.", bookClass, bookMethodName));
+
+        Object output = book.callMethod(bookMethodName, new String[]{"public"}, bookInstance);
+        assertEquals(count + incrementAmount, output,
+                String.format(incorrectIncrementBookCount, bookMethodName, this.count));
+        assertEquals(count + incrementAmount, book.getFieldValue(bookInstance, "count"),
+                String.format(incorrectIncrementBookCount, bookMethodName, this.count));
+    }
+
+    /**
+     * Bookcase tests
+     **/
+    @Test
+    public void bookcaseClassHasCorrectAttributes() {
+        String missingAttributeMessage = "Your " + bookcaseClass + " class is missing the %s attribute.";
+        String incorrectAttributeTypeMessage = "The %s attribute in your " + bookClass + " class has the wrong type.";
+        String incorrectAttributeModifierMessage = "The %s attribute in your " + bookcaseClass + " class has an incorrect modifier.";
+
+        assertTrue(bookcase.hasField(sizeCapacity), String.format(missingAttributeMessage, sizeCapacity));
+        assertTrue(bookcase.hasField(sizeCapacity, int.class), String.format(incorrectAttributeTypeMessage, sizeCapacity));
+        assertTrue(bookcase.hasModifier(sizeCapacity, "private"), String.format(incorrectAttributeModifierMessage, sizeCapacity));
+
+        try {
+            Book b = new Book(5, "fiction");
+        } catch (Error e) {
+            fail("Your program does not contain the " + bookClass + " class.");
+        }
```

Comment:
This will not work. If a class does not exist, Java fails to compile the code and throws an error. Instead, you can get the type of Book by using the getObjectClass() on the Book ObjectTest object you made in the setup method.